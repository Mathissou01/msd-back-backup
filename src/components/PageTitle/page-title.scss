@use "src/styles/01-settings/settings.colors" as colors;

.c-PageTitle {
  display: flex;
  flex-direction: column;
  gap: 16px;
  align-items: center;
  margin-top: 53px;
  padding: 20px 182px 40px;

  &__Title {
    color: colors.$expert-blue;
  }

  &__Description {
    color: colors.$gray-600;
    font-weight: 400;
    font-size: 16px;
    font-family: DINPro-Regular, serif;
    line-height: 20px;
    text-align: center;
  }
}
