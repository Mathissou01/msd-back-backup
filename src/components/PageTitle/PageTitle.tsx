import "./page-title.scss";

interface IPageTitleProps {
  title: string;
  description: string;
}

export default function PageTitle({ title, description }: IPageTitleProps) {
  return (
    <hgroup className="c-PageTitle">
      <h1 data-testid="title">{title}</h1>
      <p className="c-PageTitle__Description" data-testid="description">
        {description}
      </p>
    </hgroup>
  );
}
