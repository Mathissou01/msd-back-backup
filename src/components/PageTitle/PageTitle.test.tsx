import { render, screen } from "@testing-library/react";
import PageTitle from "./PageTitle";

const props = {
  title: "page title",
  description: "page description.",
};

describe("PageTitle", () => {
  it("renders", () => {
    const { container } = render(
      <PageTitle title={props.title} description={props.description} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("has a title and description", () => {
    render(<PageTitle title={props.title} description={props.description} />);
    const title = screen.getByTestId("title");
    expect(title).toBeInTheDocument();
    expect(title).toHaveTextContent(props.title);
    const description = screen.getByTestId("description");
    expect(description).toBeInTheDocument();
    expect(description).toHaveTextContent(props.description);
  });
});
