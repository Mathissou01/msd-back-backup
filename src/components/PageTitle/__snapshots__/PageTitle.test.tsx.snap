// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders 1`] = `
<div>
  <hgroup
    class="c-PageTitle"
  >
    <h1
      data-testid="title"
    >
      page title
    </h1>
    <p
      class="c-PageTitle__Description"
      data-testid="description"
    >
      page description.
    </p>
  </hgroup>
</div>
`;
